baybe/surrogates/base.py

```diff
@@ -56,13 +59,39 @@ class Surrogate(ABC, SerialMixin):
     """Class variable encoding whether or not the surrogate supports transfer
     learning."""
 
+    _input_transform = field(init=False, default=None, eq=False)
```

Comment:
Is there a reason for the following two fields having no type hint at all? Not even `Callable`?

Reply:
There used to be a reason in an intermediate version where tensors were involved, causing known trouble with cattrs + lazy loading. But now it's only about dataframes, so I added the hints 👍🏼